src/server.h

```diff
@@ -864,6 +864,12 @@ typedef struct moduleValue {
     void *value;
 } moduleValue;
 
+typedef enum {
+    MODULE_DEFAULT_CONFIG = 0x1,
+    MODULE_NON_DEFAULT_CONFIG = 0x2,
+    MODULE_CONFIGURED = MODULE_DEFAULT_CONFIG | MODULE_NON_DEFAULT_CONFIG
+} ModuleConfigFlags;
```

Comment:
when looking at this before reading the code, i had a hard time to understand what these mean. let's document them.
how about 
```suggestion
/* Describe the state of the module during loading, and the indication which configs were loaded / applied already. */
typedef enum {
    MODULE_CONFIGS_DEFAULTS = 0x1, /* The registered defaults were applied. */
    MODULE_CONFIGS_USER_VALS  = 0x2, /* The user provided values were applied. */
    MODULE_CONFIGS_ALL_APPLIED = 0x3 /* Both of the above applied. */
} ModuleConfigsApplied;
```